cognite/extractorutils/unstable/core/checkin_worker.py

```diff
@@ -0,0 +1,359 @@
+        self._logger.debug("Received check-in response: %s", checkin_response)
+
+        if checkin_response.last_config_revision is not None:
+            if self._active_revision == "local":
+                self._logger.warning(
+                    "Remote config revision changed "
+                    f"{self._active_revision} -> {checkin_response.last_config_revision}. "
+                    "The extractor is currently using local configuration and will need to be manually restarted "
+                    "and configured to use remote config for the new config to take effect.",
+                )
+            elif self._active_revision < checkin_response.last_config_revision:
+                self._logger.info(
+                    f"Remote config changed from {self._active_revision} to {checkin_response.last_config_revision}."
+                )
+                self._on_revision_change(checkin_response.last_config_revision)
+
+    def flush(self, cancellation_token: CancellationToken) -> None:
+        """
+        Flush available check-ins.
+
+        Arguments:
+        cancellation_token: A token to cancel the check-in reporting.
+        """
+        with self._flush_lock:
+            self._logger.debug(
+                "Going to report check-in with %d errors and %d task updates.",
+                len(self._errors),
+                len(self._task_updates),
+            )
+            self.report_checkin(cancellation_token)
+
+    def report_checkin(self, cancellation_token: CancellationToken) -> None:
+        """
+        Report a check-in to the Integrations API.
+
+        Arguments:
+        cancellation_token: A token to cancel the check-in reporting.
+        """
+        with self._start_lock:
+            if not self._has_reported_startup:
+                new_errors = [error for error in self._errors.values() if error.task is None]
+                if len(new_errors) == 0:
+                    self._logger.info("No startup request has been reported yet, skipping check-in.")
+                    return
+
+                self._logger.warning(
+                    "Check-in worker has not reported startup yet, only reporting errors not associated with a task."
+                )
+                for error in new_errors:
+                    del self._errors[error.external_id]
+                task_updates: list[TaskUpdate] = []
+            else:
+                new_errors = list(self._errors.values())
+                self._errors.clear()
+                task_updates = self._task_updates[:]
+                self._task_updates.clear()
+
+            new_errors.sort(key=lambda e: e.end_time or e.start_time)
+            task_updates.sort(key=lambda t: t.timestamp)
+
+        while not cancellation_token.is_cancelled:
+            if len(new_errors) <= MAX_ERRORS_PER_CHECKIN and len(task_updates) <= MAX_TASK_UPDATES_PER_CHECKIN:
+                self._logger.debug("Writing check-in with no batching needed.")
```

Comment:
I think this log is misleading, since it will be hit if it's the _last_ batch being sent.